bgstally/state.py

```diff
@@ -37,14 +37,11 @@ def load(self):
         self.DiscordBGSTWAutomatic:tk.StringVar = tk.StringVar(value=config.get_str('BGST_DiscordBGSTWAutomatic', default=CheckStates.STATE_OFF))
 
         self.ColonisationMaxCommodities:tk.StringVar = tk.StringVar(value=config.get_str('BGST_ColonisationMaxCommodities', default="20"))
+        self.EnableProgressScrollbar:tk.StringVar = tk.StringVar(value=config.get_str('BGST_EnableProgressScrollbar', default=CheckStates.STATE_OFF))
         self.ColonisationRCAPIKey:tk.StringVar = tk.StringVar(value=config.get_str('BGST_ColonisationRCAPIKey', default=""))
 
-        self.FcSellingCommodities:tk.StringVar = tk.StringVar(value=config.get_str('BGST_FcSellingCommodities', default=CheckStates.STATE_ON))
-        self.FcBuyingCommodities:tk.StringVar = tk.StringVar(value=config.get_str('BGST_FcBuyingCommodities', default=CheckStates.STATE_ON))
-        self.FcSellingMaterials:tk.StringVar = tk.StringVar(value=config.get_str('BGST_FcSellingMaterials', default=CheckStates.STATE_ON))
-        self.FcBuyingMaterials:tk.StringVar = tk.StringVar(value=config.get_str('BGST_FcBuyingMaterials', default=CheckStates.STATE_ON))
-        self.FcCargo:tk.StringVar = tk.StringVar(value=config.get_str('BGST_FcCargo', default=CheckStates.STATE_ON))
-        self.FcLocker:tk.StringVar = tk.StringVar(value=config.get_str('BGST_FcLocker', default=CheckStates.STATE_ON))
+        self.FcCargo:tk.StringVar = tk.StringVar(value=config.get_str('BGST_FcCargo', default="Both"))
```

Comment:
You're changing the datatype of an existing state / config field here. Have you fully protected the code against the old possible values of this field?

I have always tended to avoid changing the datatype of things that are in existing state for older versions, instead creating a new config item with a new name and discarding the old one.  That tends to be a safer approach, but if you're happy it's fully backward compatible then that's fine.

Reply:
If you're happy, I'm happy :)